src/runtime/cuda.cpp

```diff
@@ -104,8 +104,7 @@ WEAK int halide_cuda_acquire_context(void *user_context, CUcontext *ctx, bool cr
     // If the context has not been initialized, initialize it now.
     halide_assert(user_context, &context != NULL);
 
-    CUcontext local_val;
-    __atomic_load(&context, &local_val, __ATOMIC_ACQUIRE);
+    CUcontext local_val = context;
```

Comment:
I think the idea was to have the acquire here to pair with the release when the lock is released in halide_cuda_device_release . One can argue that this is not necessary as we are not locking anything here so it could race anyway -- effectively halide_cuda_device_release must only be called when no other thread is using CUDA. I guess the rule would be that the client has to guarantee some sort of acquire barrier got hit for some reason before getting here if they ever call halide_cuda_device_release. That's cagey, but not really any cagier than the design requiring the client to promise not to call halide_cuda_device_release unless nothing is using Halide's cuda support that time.